comp/softwareinventory/impl/inventorysoftware.go

```diff
@@ -161,18 +169,11 @@ func newWithClient(reqs Requires, client sysProbeClient) (Provides, error) {
 
 	is.jitter = time.Duration(localRand.Intn(max(reqs.Config.GetInt("software_inventory.jitter"), 60))) * time.Second
 	is.interval = time.Duration(max(reqs.Config.GetInt("software_inventory.interval"), 10)) * time.Minute
+	// Configure initial delay with default of 5 seconds to avoid race condition with System Probe startup
+	is.initialDelay = time.Duration(max(reqs.Config.GetInt("software_inventory.initial_delay"), 5)) * time.Second
```

Comment:
❓ question

system-probe client interface already has delay/retry facilities, are we able to reuse those?

https://github.com/DataDog/datadog-agent/blob/5758e36c0970076f9d8ef6f17d595db8eb840ef2/pkg/system-probe/api/client/client.go#L25-L28

Reply:
Yeah, I think it's up to the caller to handle it, but I think this would be a cleaner approach - maybe have an exponential backoff? Then, we also wouldn't need to expose this as a configuration.

Also, with the original approach - can we guarantee that the initial collection would work after 5 seconds?